drivers/mipi_dsi/Kconfig.stm32

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) 2023 bytes at work AG
+# SPDX-License-Identifier: Apache-2.0
+
+config MIPI_DSI_STM32
+	bool "STM32 MIPI-DSI Host Controller"
+	default y
+	depends on DT_HAS_ST_STM32_MIPI_DSI_ENABLED
+	select USE_STM32_HAL_DSI
+	select STM32_LTDC
```

Comment:
you're selecting a symbol that falls below DISPLAY, feels wrong. You likely want your driver to depend on `DT_HAS_ST_STM32_LTDC_ENABLED`. Unfortunately, our display API is a toy API, and the fact we have LTDC there is something that one day has to be fixed with proper framebuffer support.